golang/README.md

```diff
@@ -1,15 +1,65 @@
 # Golang at Clever
 
+## Onboarding
+
+It is suggested that you introduce yourself to Golang by first completing [the tour](https://tour.golang.org/welcome/1).
+Most topics are covered to help you hit the ground running.
+
+Most Go idioms can then be learned by reading [Effective Go]Effective Go](https://golang.org/doc/effective_go.html).
+This is highly suggested before you begin writing Go code for production.
+
+[The Go Blog](http://blog.golang.org/) contains many useful articles going over how to use several essential Go features like go-routines, JSON, constants, etc.
+
+
 ## Golang versions
 
-All golang services (with some exceptions, like Heka) at Clever should be on go-1.5.
+All golang services written by Clever should be on Go 1.5.
+Some 3rd-party tools that we use, such as Heka, are not on Go 1.5.
+
+
+## Best Practices
+
+### Sets
+
+Sets are typically best represented by a `map[key type]bool` structure where every boolean value is set to `true`.
```

Comment:
So if we're in a memory constrained environment the merit of using [an empty struct](http://dave.cheney.net/2014/03/25/the-empty-struct) is obviously the right move, but that is never really our case. IMO a comment explaining how the map is meant to be used is worth the improvement in dev UX.

I think the benefit is much more tangible in an `if` case where there is > 1 values being checked. Here's an example from Aeris: https://github.com/Clever/aeries/pull/108/files#diff-b06bec158b47475dafd35faa56ca4980R243

previous `if _, isException := httpsExceptions[system.ID]; parsedURL.Scheme == "http" && !isException {	`
updated: `if parsedURL.Scheme == "http" && !httpsExceptions[system.ID] {`

(in retrospect renaming `httpsExceptions` to `isHTTPSException` would've been a good move too)